xdsl/irdl/constraints.py

```diff
@@ -256,23 +199,16 @@ def verify(self, attr: Attribute, constraint_context: ConstraintContext) -> None
             raise VerifyException(f"attribute {attr} expected to be a TypedAttribute")
         self.type_constraint.verify(attr.get_type(), constraint_context)
 
-    @dataclass(frozen=True)
-    class _Extractor(VarExtractor[Attribute]):
-        inner: VarExtractor[Attribute]
-
-        def extract_var(self, a: Attribute) -> ConstraintVariableType:
-            if not isinstance(a, TypedAttribute):
-                raise PyRDLError(f"Inference expected {a} to be a TypedAttribute")
-            return self.inner.extract_var(a.get_type())
-
-    def get_variable_extractors(self) -> dict[str, VarExtractor[Attribute]]:
-        return merge_extractor_dicts(
-            self.attr_constraint.get_variable_extractors(),
-            {
-                v: self._Extractor(r)
-                for v, r in self.type_constraint.get_variable_extractors().items()
-            },
-        )
+    def variables(self) -> dict[str, ConstraintVarType]:
+        return self.type_constraint.variables() | self.attr_constraint.variables()
```

Comment:
Note that this isn't doing the duplicate key checking that the extractor implementation was doing.

Reply:
If we merge this version it would be nice to have it

Reply:
Am I right in understanding that this is no longer relevant?